File: app/src/main/java/com/widyatama/nurseassistant/features/measurement/MeasurementActivity.java
package com.widyatama.nurseassistant.features.measurement;

import android.os.Bundle;

import com.widyatama.core.base.BaseActivity;
import com.widyatama.core.util.SchedulerProviderUtil;
import com.widyatama.nurseassistant.R;

import io.reactivex.disposables.CompositeDisposable;

public class MeasurementActivity extends BaseActivity implements MeasurementViewContracts {

    private MeasurementPresenter<MeasurementViewContracts> presenter;

    @Override
    protected void onInitializedView(Bundle savedInstanceState) {
        presenter = new MeasurementPresenter<MeasurementViewContracts>(new SchedulerProviderUtil(),
            new CompositeDisposable());
        presenter.onAttach(this);
    }

    @Override
    protected void onDestroy() {
        presenter.onDetach();
        super.onDestroy();
    }

    @Override
    protected Integer setLayout() {
        return R.layout.activity_measurement;
    }
}

File: app/src/main/java/com/widyatama/nurseassistant/features/measurement/MeasurementActivity.kt
package com.widyatama.nurseassistant.features.measurement

import android.os.Bundle
import com.widyatama.core.base.BaseActivity
import com.widyatama.nurseassistant.R
import org.koin.android.ext.android.inject

class MeasurementActivity : BaseActivity(), MeasurementViewContracts {

    val presenter: MeasurementPresenter<MeasurementViewContracts> by inject()

    override fun onInitializedView(savedInstanceState: Bundle?) {
        presenter.onAttach(this)
    }

    override fun onDestroy() {
        presenter.onDetach()
        super.onDestroy()
    }

    override fun setLayout(): Int {
        return R.layout.activity_measurement
    }
}
